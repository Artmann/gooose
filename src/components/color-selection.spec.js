import ColorSelection from "./color-selection";
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('ColorSelection', () => {
  it('calls onChange when you click a color', () => {
    const onChange = jest.fn();
    const wrapper = mount(<ColorSelection color='#2779bd' onChange={ onChange } />);

    wrapper
      .find('[data-test-color="#eb5286"]')
      .first()
      .simulate('click');

    expect(onChange).toHaveBeenCalledWith('#eb5286');
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<ColorSelection />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a selected color', () => {
    const tree = renderer
      .create(<ColorSelection color='#2779bd' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a random color', () => {
    const tree = renderer
      .create(<ColorSelection color='red' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
