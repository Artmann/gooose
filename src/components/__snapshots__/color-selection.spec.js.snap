// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelection renders correctly 1`] = `
<div
  className="sc-bdVaJa ieBYTJ"
>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#eb5286"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#eb5286",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#794acf"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#794acf",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#5661b3"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#5661b3",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#2779bd"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#2779bd",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#38a89d"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#38a89d",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#1f9d55"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#1f9d55",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#f2d024"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#f2d024",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#de751f"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#de751f",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorSelection renders correctly with a random color 1`] = `
<div
  className="sc-bdVaJa ieBYTJ"
>
  <div
    className="sc-bwzfXH jgELGV"
    data-test-color="red"
    onClick={[Function]}
    selected={true}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "red",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#eb5286"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#eb5286",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#794acf"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#794acf",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#5661b3"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#5661b3",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#2779bd"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#2779bd",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#38a89d"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#38a89d",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#1f9d55"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#1f9d55",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#f2d024"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#f2d024",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#de751f"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#de751f",
        }
      }
    />
  </div>
</div>
`;

exports[`ColorSelection renders correctly with a selected color 1`] = `
<div
  className="sc-bdVaJa ieBYTJ"
>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#eb5286"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#eb5286",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#794acf"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#794acf",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#5661b3"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#5661b3",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH jgELGV"
    data-test-color="#2779bd"
    onClick={[Function]}
    selected={true}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#2779bd",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#38a89d"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#38a89d",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#1f9d55"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#1f9d55",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#f2d024"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#f2d024",
        }
      }
    />
  </div>
  <div
    className="sc-bwzfXH fzDxln"
    data-test-color="#de751f"
    onClick={[Function]}
    selected={false}
  >
    <div
      className="sc-htpNat fCGVby"
      style={
        Object {
          "background": "#de751f",
        }
      }
    />
  </div>
</div>
`;
